modules/cache-common/src/main/java/org/opensearch/cache/common/tier/TieredSpilloverCache.java

```diff
@@ -156,10 +165,10 @@ public void invalidateAll() {
      * Provides an iteration over both onHeap and disk keys. This is not protected from any mutations to the cache.
      * @return An iterable over (onHeap + disk) keys
      */
-    @SuppressWarnings("unchecked")
+    @SuppressWarnings({ "rawtypes", "unchecked" })
     @Override
     public Iterable<K> keys() {
-        return Iterables.concat(onHeapCache.keys(), diskCache.keys());
+        return new ConcatenatedIterables<K>(new Iterable[] { onHeapCache.keys(), diskCache.keys() });
```

Comment:
Can this be `new Iterable<K>[]` so that you don't need the `rawtypes` suppression?

Reply:
That didn't work ie using `new Iterable<K>[]`. But I have changed it in a way to avoid rawTypes suppression.